### ISSUE TITLE
GoCD became extremely slow and not responsive periodically

### ISSUE THREAD

**Issue by victor-sim** [gocd/gocd#4168]
##### Issue Type
- Bug Report

##### Summary

Almost once per every one or two days, GoCD becomes extremely slow and not responsive.

##### Environment
8 Virtual Processors
10 GB RAM
7 GB XMX

###### Basic environment details

* Go Version: 17.11.0 (5520-9f6909e2f64b07d2dce5cecd4ea5b92b8e19d6b1)
* JAVA Version: 1.8.0_102
* OS: Windows Server 2012 R2 6.3
* Browser vendor and version (if relevant): Not relevant

###### Additional Environment Details
[gocd_api_support.txt](https://github.com/gocd/gocd/files/1612496/gocd_api_support.txt)

[290.txt](https://github.com/gocd/gocd/files/1612551/290.txt)

##### Steps to Reproduce

1. Keep running GoCD for one or two days
2. GoCD became slowdown and almost not responsive at all
3. It never goes back to normal unless restarting GoCD service.
4. When it became slow, agents fail to fetch artifacts, lost connection, and elastic agents are failing auto registration.

##### Any other info
Whenever GoCD extremely slows down, a total number of DiskSpaceChecker threads increased dramatically. (eg. ArtifactsDiskCleaner class, ArtifactsDiskSpaceFullChecker class, DatabaseDiskSpaceFullChecker class...)

When a GoCD status is okay there are 2-3 threads that are run DiskSpaceChecker.check(). But there are more than 100 threads when GoCD is slow and not responsive and CPU time of these threads are several hundred seconds.

**Comment by agalpin** [comment]
We are having a similar issue, running 17.12 on linux under an alpine docker container, we have to restart physical machine as just restarting container doesn't help!

**Comment by victor-sim** [comment]
It seems like heavy load of Public API caused this issue.

We used our own dashboard that shows a historical view of value stream maps, and it reads job properties values at least 700 times per each minute to show custom job property values on the custom dashboard.
When I doubled the number of jobs that require properties in dashboard (so that means the number of GetJobProperties API, GET /go/properties/:pipeline_name/:pipeline_counter/:stage_name/:stage_counter/:job_name, call became doubled) frequency of unresponsive GoCD became almost doubled.

And when I restricted the number of GetJobProperties API call to 50 calls per minute, this stop the world issue never happened for last 4 days.
